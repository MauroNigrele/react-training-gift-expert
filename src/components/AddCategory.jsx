import { useState } from "react"
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory, currentCategories }) => {
    
    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({target}) => {
        setInputValue(target.value)
    }

    const onFormSubmit = (event) => {
        event.preventDefault();
        if(inputValue.trim().length <= 1 ) return;
        onNewCategory(inputValue.trim());
        setInputValue('');
    }

    return (
    <form onSubmit={ onFormSubmit } aria-label="form">
        <input 
            type="text" 
            placeholder="Gifs Search"
            value={ inputValue }
            onChange={ onInputChange }
            // onChange={ (event) => onInputChange(event) } // Same Result 
        />
    </form>
  )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}