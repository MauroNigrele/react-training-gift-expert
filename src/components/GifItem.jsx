import PropTypes from 'prop-types';
import React from 'react'

export const GifItem = ({title, url}) => {
    return (
        <div className='card'>
            <h4> {title} </h4>
            <img src={ url } alt={ title }/>
        </div>
    )
}

GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}