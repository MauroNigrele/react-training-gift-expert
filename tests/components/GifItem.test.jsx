import { render, screen, fireEvent } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem";


describe('Tests for GifItem Component', () => { 

    const testTitle = 'Google Page';
    const testUrl = 'https://www.google.com/';

    test('should match with snapshot', () => {     
        const {container} = render( <GifItem title={ testTitle } url={ testUrl } /> );
        expect(container).toMatchSnapshot();
    });

    test('should display image url and alt', () => {     
        render( <GifItem title={ testTitle } url={ testUrl } /> );
        // console.log(screen.debug());
        // console.log(screen.getByRole('img').src);
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(testUrl);
        expect(alt).toBe(testTitle);
    });
    
    test('should display title', () => {     
        render( <GifItem title={ testTitle } url={ testUrl } /> );
        expect(screen.getByText( testTitle )).toBeTruthy();
    });





    

});
