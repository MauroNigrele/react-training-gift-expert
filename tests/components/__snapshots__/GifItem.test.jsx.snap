// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for GifItem Component should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <h4>
       
      Google Page
       
    </h4>
    <img
      alt="Google Page"
      src="https://www.google.com/"
    />
  </div>
</div>
`;
